client/app/services/query.js

```diff
@@ -24,44 +34,75 @@ class Parameter {
     this.title = parameter.title;
     this.name = parameter.name;
     this.type = parameter.type;
-    this.value = parameter.value;
+    this.useCurrentDateTime = parameter.useCurrentDateTime;
     this.global = parameter.global;
     this.enumOptions = parameter.enumOptions;
     this.queryId = parameter.queryId;
 
-    // method to update parameter value from date/time picker component
-    // (react does not support two-way binding with `ngModel`)
-    this.updateValue = (value) => {
-      this.ngModel = value;
-    };
+    // validate value and init internal state
+    this.setValue(parameter.value);
+
+    // explicitly bind it to `this` to allow passing it as callback to Ant's
+    // DatePicker component
+    this.setValue = this.setValue.bind(this);
   }
 
-  get ngModel() {
-    if (this.type === 'date' || this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
-      this.$$value = this.$$value || moment(this.value).toDate();
-      return this.$$value;
-    } else if (this.type === 'number') {
-      this.$$value = this.$$value || parseInt(this.value, 10);
-      return this.$$value;
-    }
+  get isEmptyValue() {
+    return isNull(this.value) || isUndefined(this.value) || (this.value === '');
+  }
 
+  get isEmpty() {
+    // `isEmptyValue` may return true, but parameter may actually be not empty -
+    // in example, date/time params may have default value
+    return isNull(this.getValue());
+  }
+
+  getValue() {
+    if (this.isEmptyValue) {
+      if (
+        includes(['date', 'datetime-local', 'datetime-with-seconds'], this.type) &&
+        this.useCurrentDateTime
+      ) {
+        return moment().format(DATETIME_FORMATS[this.type]);
+      }
+      return null; // normalize empty value
+    }
+    if (this.type === 'number') {
+      return parseInt(this.value, 10) || null; // normalize empty value
```

Comment:
Want to guess what will `parseInt(0, 10) || null` return? :)

Reply:
Oooops... thanks you, will fix